File: BOJ/1406/main.py
import sys
readline = sys.stdin.readline

stack_l = [c for c in readline().strip()]
stack_r = []

M = int(readline())
for _ in range(M):
  command = readline().strip()
  if command == "L":
    if stack_l: stack_r.append(stack_l.pop())
  elif command == "B":
    if stack_l: stack_l.pop()
  elif command == "D" :
    if stack_r: stack_l.append(stack_r.pop())
  else:
    stack_l.append(command[2])

while stack_r:
  stack_l.append(stack_r.pop())

sys.stdout.write("".join(stack_l))

File: BOJ/1406/main.js
let line = 0;
const input = require("fs")
  .readFileSync(process.platform === "linux" ? "/dev/stdin" : "./BOJ/1406/input.txt")
  .toString()
  .split("\n");

const stack_l = Array.from(input[line++]);
const stack_r = [];
const M = input[line++];

while (line < input.length) {
  const command = input[line++];
  switch (command) {
    case "L":
      if (stack_l.length) stack_r.push(stack_l.pop());
      break;
    case "D":
      if (stack_r.length) stack_l.push(stack_r.pop());
      break;
    case "B":
      if (stack_l.length) stack_l.pop();
      break;
    default:
      stack_l.push(command[2]);
      break;
  }
}
while (stack_r.length > 0) stack_l.push(stack_r.pop());

console.log(stack_l.join(""));
